torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -533,6 +533,9 @@ def __call__(self, kernel_idx, constant_args_idx, grid, kwargs):
             kwargs=kwargs,
         )
 
+    def pure(self):
+        return False
```

Comment:
Suggestion: could we add a "*, pure=False" option to HigherOrderOperator's ctor https://github.com/pytorch/pytorch/blob/b6d6aa49b839b11c6bb01688975d2ef77c3936d0/torch/_ops.py#L248 and then have HigherOrderOperator's `def pure(self)` pull from there? This way we make it obvious to HOP authors that this is an option they can set when defining their HOP class.